src/dda/utils/git/changeset.py

```diff
@@ -58,78 +60,45 @@ class ChangeSet:  # noqa: PLW1641
     When considering the changes to the working directory, the untracked files are considered as added files.
     """
 
-    def __init__(self, changes: dict[Path, ChangedFile]) -> None:
-        self.__changes = MappingProxyType(changes)
+    def __init__(self, changed_files: Iterable[ChangedFile]) -> None:
+        self.__changed = MappingProxyType({str(c.path): c for c in changed_files})
+        self.__files = tuple(self.__changed.values())
 
     @property
-    def changes(self) -> MappingProxyType[Path, ChangedFile]:
-        return self.__changes
+    def paths(self) -> MappingProxyType[str, ChangedFile]:
```

Comment:
This provides the previous interface of all changes being represented by a mapping of relative paths to the ChangedFile class and removes the requirement for pass-through dictionary methods. The keys are now strings because msgspec actually doesn't [support](https://jcristharif.com/msgspec/supported-types.html#dict) any other types for dictionary keys which is why the decoding hook had to manually coerce the type.

A benefit to this is that callers possessing `str` instances no longer have to import the `Path` class.